boost_histogram/_internal/hist.py

```diff
@@ -406,5 +405,5 @@ def __getitem__(self, index):
             )
 
     def __setitem__(self, index, value):
-        indexes = _compute_commonindex(self._hist, index, expand=True)
+        indexes = _compute_commonindex(self._hist, index, value)
```

Comment:
Note to self: See why this changed...

Reply:
This should be False for now. Later, when we support setting with arrays, this will be True again.

It never should be "value"...

Reply:
Mostly, I was puzzled by the name of the function and I didn't look it up. I think this should be combined with `_at_set`, to make it faster. But that is another PR.

Reply:
Allocating a temporary array for the linearized indices is not necessary and costs significant performance.

Reply:
First we should make everything work, then we can optimize. :)